drizzlepac/alignimages.py

```diff
@@ -0,0 +1,1075 @@
+
+# Module-level dictionary contains instrument/detector-specific parameters used later on in the script.
+detector_specific_params = {"acs":
+                                {"hrc":
+                                     {"fwhmpsf": 0.073,
+                                      "classify": True,
+                                      "threshold": None},
+                                 "sbc":
+                                     {"fwhmpsf": 0.065,
+                                      "classify": False,
+                                      "threshold": 2.0},
+                                 "wfc":
+                                     {"fwhmpsf": 0.13, #0.076,
+                                      "classify": True,
+                                      "threshold": -1.1}},
+                            "wfc3":
+                                {"ir":
+                                     {"fwhmpsf": 0.14,
+                                      "classify": False,
+                                      "threshold": None},
+                                 "uvis":
+                                     {"fwhmpsf": 0.076,
+                                      "classify": True,
+                                      "threshold": None}}} # fwhmpsf in units of arcsec
+
+
+# ----------------------------------------------------------------------------------------------------------------------
+
+
+def check_and_get_data(input_list,**pars):
+    """Verify that all specified files are present. If not, retrieve them from MAST.
+
+    Parameters
+    ----------
+    imglist : list
+        List of one or more calibrated fits images that will be used for catalog generation.
+
+    Returns
+    =======
+    input_file_list : list
+        list of full filenames
+
+    """
+    totalInputList = []
+    retrieve_list = [] # list of already retrieved ASNs, only request each once
+    for input_item in input_list:
+        filelist = aqutils.retrieve_observation(input_item,**pars)
+        if len(filelist) == 0:
+            # look for local copy of the file
+            input_rootname = input_item[:input_item.find('_')]
+            fitsfilenames = sorted(glob.glob("{}_fl?.fits".format(input_rootname)))
+            if len(fitsfilenames) > 0:
+                imghdu = fits.open(fitsfilenames[0])
+                imgprimaryheader = imghdu[0].header
+                try:
+                    asnid = imgprimaryheader['ASN_ID'].strip().lower()
+                    if asnid == 'NONE':
+                        asnid = None
+                except KeyError:
+                    asnid = None
+                if asnid and asnid not in retrieve_list:
+                    filelist = aqutils.retrieve_observation(asnid,**pars)
+                    retrieve_list.append(asnid) # record asn's already retrieved
+
+        if len(filelist) > 0:
+            totalInputList += filelist
+
+    if len(filelist) > 0: totalInputList = sorted(
```

Comment:
@mdlpstsci This code crashes when input list is empty because `filelist` is undefined. Also, is it really important to check the length of the last (set in the last iteration of the loop above) `filelist` or should it rather be `totalInputList`.